### PR TITLE
Fix extractor

### LINKED ISSUE #22461: Support of Le Monde
## Checklist

- [X] I'm reporting a new site support request
- [X] I've verified that I'm running youtube-dl version **2019.09.12.1**
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that none of provided URLs violate any copyrights
- [X] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single video: https://www.lemonde.fr/festival/video/2017/08/30/que-se-passerait-il-si-tout-le-monde-etait-vegan_5178386_4415198.html

## Description

WRITE DESCRIPTION HERE

Le Monde videos are actually on YouTube but youtube-dl doesn't catch them.

### PR DESCRIPTION
**Pull request by LucBerge**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

LeMonde has changed its website. Videos are now hosted on youtube or dailymotion. Closes #22461.

### CODE REVIEW

File: youtube_dl/extractor/lemonde.py [hunk 1]
```diff
@@ -1,58 +1,64 @@
...
+        "youtube": "https://www.youtube.com/embed/{}",
+        "dailymotion": "https://www.dailymotion.com/video/{}"
+    }
+
     def _real_extract(self, url):
         display_id = self._match_id(url)
-
         webpage = self._download_webpage(url, display_id)
 
-        digiteka_url = self._proto_relative_url(self._search_regex(
```
Comment (dstftw) [inline youtube_dl/extractor/lemonde.py]: Do not touch this.

File: youtube_dl/extractor/lemonde.py [hunk 2]
```diff
@@ -1,58 +1,64 @@
...
-            r'url\s*:\s*(["\'])(?P<url>(?:https?://)?//(?:www\.)?(?:digiteka\.net|ultimedia\.com)/deliver/.+?)\1',
-            webpage, 'digiteka url', group='url', default=None))
+        mobj = re.search(r'data-id="(?P<id>.+)" data-provider="(?P<provider>[^ ]+)"', webpage)
+        video_id = mobj.group('id')
+        provider = mobj.group('provider')
+
+        if(provider not in self._PROVIDERS):
+            raise ExtractorError('Unsupported provider ' % provider)
```
Comment (dstftw) [inline youtube_dl/extractor/lemonde.py]: Remove.

File: youtube_dl/extractor/lemonde.py [hunk 3]
```diff
@@ -1,58 +1,64 @@
...
 
-        if digiteka_url:
-            return self.url_result(digiteka_url, 'Digiteka')
+        embeded_url = self._PROVIDERS.get(provider, "").format(video_id)
 
-        return self.url_result(url, 'Generic')
+        return {
+            '_type': 'url',
+            'url': embeded_url
```
Comment (dstftw) [inline youtube_dl/extractor/lemonde.py]: Add `id` and `ie_key`.

File: youtube_dl/extractor/lemonde.py [hunk 4]
```diff
@@ -1,58 +1,64 @@
...
+        provider = mobj.group('provider')
+
+        if(provider not in self._PROVIDERS):
+            raise ExtractorError('Unsupported provider ' % provider)
 
-        if digiteka_url:
-            return self.url_result(digiteka_url, 'Digiteka')
+        embeded_url = self._PROVIDERS.get(provider, "").format(video_id)
 
-        return self.url_result(url, 'Generic')
```
Comment (dstftw) [inline youtube_dl/extractor/lemonde.py]: Do not touch this.

File: youtube_dl/extractor/lemonde.py [hunk 5]
```diff
@@ -1,58 +1,64 @@
...
         display_id = self._match_id(url)
-
         webpage = self._download_webpage(url, display_id)
 
-        digiteka_url = self._proto_relative_url(self._search_regex(
-            r'url\s*:\s*(["\'])(?P<url>(?:https?://)?//(?:www\.)?(?:digiteka\.net|ultimedia\.com)/deliver/.+?)\1',
-            webpage, 'digiteka url', group='url', default=None))
+        mobj = re.search(r'data-id="(?P<id>.+)" data-provider="(?P<provider>[^ ]+)"', webpage)
+        video_id = mobj.group('id')
+        provider = mobj.group('provider')
```
Comment (dstftw) [inline youtube_dl/extractor/lemonde.py]: Must not be fatal. Must fallback to generic if not found.

File: youtube_dl/extractor/lemonde.py (hunk 1 above)
Comment (LucBerge) [reply youtube_dl/extractor/lemonde.py]: This is not up to date with the website, it always return `None`.

File: youtube_dl/extractor/lemonde.py [hunk 6]
```diff
@@ -1,58 +1,62 @@
...
     def _real_extract(self, url):
         display_id = self._match_id(url)
-
         webpage = self._download_webpage(url, display_id)
 
-        digiteka_url = self._proto_relative_url(self._search_regex(
-            r'url\s*:\s*(["\'])(?P<url>(?:https?://)?//(?:www\.)?(?:digiteka\.net|ultimedia\.com)/deliver/.+?)\1',
-            webpage, 'digiteka url', group='url', default=None))
-
+        digiteka_url = self._proto_relative_url(self._search_regex(r'url\s*:\s*(["\'])(?P<url>(?:https?://)?//(?:www\.)?(?:digiteka\.net|ultimedia\.com)/deliver/.+?)\1',webpage, 'digiteka url', group='url', default=None))
```
Comment (dstftw) [inline youtube_dl/extractor/lemonde.py]: Revert it exactly as it was.

File: youtube_dl/extractor/lemonde.py [hunk 7]
```diff
@@ -1,58 +1,62 @@
...
         if digiteka_url:
             return self.url_result(digiteka_url, 'Digiteka')
 
-        return self.url_result(url, 'Generic')
+        mobj = re.search(r'data-id="(?P<id>.+)" data-provider="(?P<provider>[^ ]+)"', webpage)
+        video_id = mobj.group('id') if mobj else display_id
```
Comment (dstftw) [inline youtube_dl/extractor/lemonde.py]: Incorrect.

File: youtube_dl/extractor/lemonde.py [hunk 8]
```diff
@@ -1,58 +1,62 @@
...
+        provider = mobj.group('provider') if mobj else ""
+
+        embeded_url = self._PROVIDERS.get(provider, "{}").format(video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/lemonde.py]: Incorrect.

File: youtube_dl/extractor/lemonde.py (hunk 4 above)
Comment (dstftw) [reply youtube_dl/extractor/lemonde.py]: ^

File: youtube_dl/extractor/lemonde.py [hunk 9]
```diff
@@ -1,58 +1,62 @@
...
+            'timestamp': 1477044897,
+            'uploader_id': 'x1wd0c',
+            'uploader': 'Le Monde.fr',
+            'description': 'CETA. 4 lettres pour Comprehensive Economic and Trade Agreement, ou Accord économique et commercial global). Un nom un peu barbare pour un traité transatlantique, cousin du Tafta, qui agite en ce moment l’actualité européenne.\xa0Pour comprendre les enjeux de cet accord commercial, notre spécialiste Maxime Vaudano.'
+        }
     }]
 
+    _PROVIDERS = {
+        "youtube": "https://www.youtube.com/embed/{}",
+        "dailymotion": "https://www.dailymotion.com/video/{}"
```
Comment (dstftw) [inline youtube_dl/extractor/lemonde.py]: Single quotes. No `{}` here.

**Pull request closed without merging**